Documentation/tuning.md

```diff
@@ -0,0 +1,73 @@
+
+## Recommended values
+
+Besides the default limits example, we also provide guidelines and examples of what a preferred set of values would look like on a larger cluster.
+
+Here are a few guidelines:
+
+- Presto, hive, and hdfs are all written in Java, and thus tend to consume more memory than other applications.
+- Metering is naturally batch oriented and can be bursty in resources, and thus there are many idle periods. Without proper autoscaling, it may be necessary to over-provision resources if you want optimal performance, or find that applications crash without more resources.
+- Metering is designed to run at even the largest scale environments like Openshift online, which has clusters with over 5000 namespaces, and over 10,000 pods. It works well on smaller clusters too, but the cost of being able to running at these larger scales can mean the metering stack isn't as easily tuned for a smaller footprint.
+
+In general, we can only provide guidelines based on the environments we run, so we always recommend using your monitoring stack in addition to our documentation to determine what resource limits make sense for you.
+As we work to improve metering, over time we hope to address more of these concerns as part of the metering-operator by leveraging Kubernetes pod autoscaling, jobs, and newer features in the various components we leverage.
+
+We recommend starting with our [recommended-resources-limit.yaml][recommended-resource-limits] example configuration, and adjusting it based on monitoring data over the period of a few weeks. It's much easier to start with higher values and lower them until you see pods restarting or when monitoring shows pods reaching their resource limits.
+
+## General Advice
+
+Below is some generally useful information to know when trying to tune the various components and understand _why_ we suggest specific values in our [recommended-resources-limit.yaml][recommended-resource-limits] example configuration.
+
+### reporting-operator
+
+The reporting-operator's most resource intensive task is querying Prometheus for metrics, storing the results in memory, converting them into a SQL insert statement, and batching those inserts to Presto.
+Performing imports is mostly a memory and IO intensive task, and thus increasing memory beyond defaults can help when dealing with a lot of metrics on larger or highly active clusters.
+
+### Presto
+
+Presto is the main component that does real computation intensive work.
+Additionally, it processes everything in memory when a query comes in, meaning it's primary resource constraint will often be memory, and CPU can be a problem when there's a lot of garbage collections occurring while processing a query.
+
+Presto itself doesn't store any data, and instead access data on demand when a query is executed.
+This means network is often a bottleneck, and can be something you may need to consider.
```

Comment:
Presto itself doesn't store the data, instead it accesses the data on demand ...

Reply:
Good catch